### DISCUSSION TITLE
UI Tweaks

### DISCUSSION THREAD

**Discussion by seerge** [seerge/g-helper#875] [General]
Hello fellow community,

I want to coordinate some UI tweaks (and ask for opinion) on main app window :)

Idea is to move "Extra" (i.e. Settings) button somewhere in more prominent, and not linked to keyboard section window + cleanup layout a bit to free up space for possible other elements

Current proposal:
![Screenshot 2023-07-24 145559](https://github.com/seerge/g-helper/assets/5920850/c621f60d-81c1-43ed-8b2c-c24e7dad153d)

Next steps would be simplifying anime-matrix section : I plan to add a separate window for that with more settings / options (i.e. option to add custom text and visual grid of few suggested gifs/pics)

Let me know what do you think / or if you have other ideas
 
[GHelper.zip](https://github.com/seerge/g-helper/files/12146129/GHelper.zip)

**Comment by ke1ne** [comment]
I would keep the Exit button text too, for the better consistency.

**Reply by seerge** [reply]
May be, but i intentionally want to make button less prominent / popping than others. As it's way less important :) 
[GHelper.zip](https://github.com/seerge/g-helper/files/12146357/GHelper.zip)

**Reply by ke1ne** [reply]
@seerge I'm OK with this variant. Thanks! As a next step, I would propose to do something with the Keyboard section. The Color control (button) has lower height than the Mode dropdown:
![image](https://github.com/seerge/g-helper/assets/8934421/b418e282-7661-46da-9128-ac231ec337c8)

**Reply by seerge** [reply]
@ke1ne problem that buttons scale differently comparing to dropdowns (heil to windows/.net scaling) and I initially aim to 200% :) 

But check 0.102 , i have made buttons a bit bigger there in general, so it may look better on 150% scaling as well.

**Reply by IceStormNG** [reply]
<img width="302" alt="image" src="https://github.com/seerge/g-helper/assets/12786283/58741fa8-98f2-40e6-86fc-582f292c1b95">

The size itself is fine, but the buttons are positioned a little higher than the drop downs for me (at 150%). It's minor though and I had to look twice to even notice that.

**Reply by seerge** [reply]
Thats also cause of scaling as for 200% they are centred :)) i need to look for some better looking dropdowns probably 

On a side note, can you make attractive screenshot of main window + mouse settings including some wallpaper (like i usually do), we can add it to new build to promote your mouse additions :)

**Reply by IceStormNG** [reply]
> On a side note, can you make attractive screenshot of main window + mouse settings including some wallpaper (like i usually do), we can add it to new build to promote your mouse additions :)

Sure. Let me just restart G-Helper in english  and do some shots. Is dark mode fine? Or do you want it in light mode like you always did?

**Reply by seerge** [reply]
It doesn’t really matter :)

**Reply by IceStormNG** [reply]
Sorry for cluttering the response here lol.
Not sure what you like better. I made 2 each. One with the mouse menu open and one with it closed but the mouse connected. I hope there is one that you like.

![Screenshot 2023-07-25 225625](https://github.com/seerge/g-helper/assets/12786283/16428b9d-eb2d-4667-b36d-d02816049f67)
![Screenshot 2023-07-25 225639](https://github.com/seerge/g-helper/assets/12786283/4cb95735-e6e6-4141-8280-5efd543c137c)
![Screenshot 2023-07-25 225739](https://github.com/seerge/g-helper/assets/12786283/3448e85a-ecf2-4c28-a7f1-70f704b3778e)
![Screenshot 2023-07-25 225750](https://github.com/seerge/g-helper/assets/12786283/2ba7bc66-a06e-4bfe-a673-b9107f60d769)
![Screenshot 2023-07-25 225818](https://github.com/seerge/g-helper/assets/12786283/5ed97172-46e1-49e1-a723-314602e2902f)
![Screenshot 2023-07-25 225832](https://github.com/seerge/g-helper/assets/12786283/64680e28-596e-48e0-beeb-97631b19a730)

**Comment by flankedgonerogue** [comment]
This is better but I'd like to emphasis that the free space under Laptop Keyboard can be filled up by a dedicated button for the keyboard settings under the current Extra window. So only the other part remains in the extra window. This does complicate (maybe clutter too) the UI a bit but separation would be worth it given the expansion of app feature set. For example, when I started using G-Helper the other section had only 3-4 check boxes now it has 8 + log + services.

**Reply by seerge** [reply]
@flankedgonerogue i'm not sure if adding million windows to the app will make UI easier. Initially whole Extra section was indeed dedicated to keyboard (that's why it was in keyboard section). I just had to add a few strange checkboxes that some people asked to add somewhere :)

**Reply by flankedgonerogue** [reply]
Yes, I agree that it will take away the simplicity.

**Comment by seerge** [comment]
I have actually digested everything, and decided not to rush even with Extra button. As it's de-facto mainly dedicated to keyboard bindings / settings atm (+ a few of minor and very optional checkboxes). I have moved it back :) but kept some other changes.

Also for G14 (and other devices with animatrix), i moved animatrix section above keyboard, so Extra is always in bottom-right part of the UI

[GHelper.zip](https://github.com/seerge/g-helper/files/12160018/GHelper.zip)

![Screenshot 2023-07-25 122800](https://github.com/seerge/g-helper/assets/5920850/b3753612-3f76-4963-b2dd-a21fa542effe)

**Reply by ke1ne** [reply]
@seerge As a slow-but-stable way, why not. Thank you

**Reply by flankedgonerogue** [reply]
Cleaner, I like it.

**Comment by alexatthehub** [comment]
I think it makes sense to divorce more general application settings from the keyboard section, as you have in the image. However I think it might be a mistake to divorce keyboard specific settings from the keyboard section. Sure, a user can memorize where it is they need to go, but in terms of discoverability, I think ultimately it makes sense for general application settings and keyboard settings to be accessed separately.

I see a few other opportunities to de-clutter or organize the UI for thematic coherence and convenience:
- "Run on startup" is kind of a set-it-and-forget-it setting and might make more sense out of the way, in general application settings.
- Software versions are usually to be found in an About window, but might make more sense in the updates window.
- Panel overdrive is neither a keyboard nor general application setting. I think I see a potentially elegant way of getting it into the Screen section; if the refresh rate itself is a drop down, that frees up a toggleable button for overdrive. This way it is no longer hidden among other extra settings, and more where one might expect to find it.

![Untitled-2](https://github.com/seerge/g-helper/assets/140471037/b816ad1c-9979-4780-8f12-3563e7f97d2f)

These are just thoughts and by no means a critique of what you've accomplished so far! This program is amazing as it is.

**Reply by flankedgonerogue** [reply]
Could I get this build to review "in-person"? or is this just a render?

**Reply by alexatthehub** [reply]
Just a render - a bit of photoshop.

On further thought, I think it's worth noting that this design would be breaking an important convention in the screen section. Auto, overdrive, and multizone are all on/off toggles in my mockup, with no clear indication that's the case from the chrome. In other words, the power and gpu modes function like radio buttons, and I break that convention in the screen section.

**Comment by thduc87** [comment]
I'd like to put the extra button to the bottom at the same level as the update and quit buttons, also run on start up check could be put inside the extra settings in my opinion and by default it is checked.

**Reply by thduc87** [reply]
also currently when close the main window of ghelper it doesn't close the extra open windows, so i'd like to suggest when close the main window then all open windows will be closed as well, thanks.

**Comment by alexatthehub** [comment]
I think the titles for the screen and keyboard areas could just be called "Screen" and "Keyboard". The designator, "laptop", seems like something we could do without. A little less text might make things a little cleaner.

**Reply by seerge** [reply]
@alexatthehub possibly, but some people have external screen - that's why specified what screen is being controlled :)

As for keyboard - my point was to underline that it's backlight controls (as dropdown with modes + color button don't say that they are backlight)

**Comment by left-winger502** [comment]
Please consider app's window to remember its position it was moved to. When the taskbar is set to auto hide (which is what many people use with oleds) it overlays some of the g-helpers bottom buttons and controls when rolled out, making them inaccessible which is kind of annoying. Thank you!
